recipes/onnx/all/conanfile.py

```diff
@@ -140,25 +147,30 @@ def _onnx_components(self):
                 "libs": ["onnx_proto"],
                 "defines": ["ONNX_NAMESPACE=onnx", "ONNX_ML=1"],
```

Comment:
(another comment not related to this PR)

Some projects:

* onnx-tensorrt https://github.com/onnx/onnx-tensorrt/blob/fa0964e8477fc004ee2f49ee77ffce0bf7f711a9/CMakeLists.txt#L94
* glow https://github.com/pytorch/glow/blob/master/lib/Importer/CMakeLists.txt#L9

Override the ONNX_NAMESPACE before building onnx to avoid symbols collisions at Protobuf registration
https://stackoverflow.com/questions/51795326/static-linking-of-protocol-buffers-results-in-conflicts-with-existing-symbols

I'm starting to believe that ONNX_NAMESPACE will have to be parametrizable via conanfile options.